src/v2/BaseLoginRouter.js

```diff
@@ -129,26 +81,18 @@ export default Router.extend({
   },
 
   handleIdxResponseSuccess (idxResponse) {
-    const { interactionCode } = idxResponse;
-    if (interactionCode) {
-      const authClient = this.settings.getAuthClient();
-      const codeVerifier = this.settings.get('codeVerifier');
-      this.settings.unset('codeVerifier');
-      return authClient.token.exchangeCodeForTokens({ codeVerifier, interactionCode })
-        .then(({ tokens }) => {
-          this.settings.callGlobalSuccess(Enums.SUCCESS, { tokens });
-        })
-        .catch(err => {
-          this.settings.callGlobalError(err);
-        });
+    if (idxResponse.interactionCode) {
+      return completeLoginFlow(this.settings, idxResponse);
     }
-
+  
     // transform response
-    const ionResponse = _.compose(
-      i18nTransformer,
-      uiSchemaTransformer,
-      responseTransformer.bind({}, this.settings),
-    )(idxResponse);
+    const ionResponse = transformIdxResponse(this.settings, idxResponse);
+
+    // if this is a terminal error, clear all transaction meta
```

Comment:
Out of curiosity, why do we want to clear all metadata when this occurs?

Reply:
To avoid trying to replay the transaction and receiving the same error. (The assumption is the error is permanent and non-recoverable) Sometimes the "signout" / "cancel" button does not appear on these error screens. In this case, reloading the page will show a fresh signin dialog rather than replaying the transaction and receiving an error again.